libvirt/utils_volume.go

```diff
@@ -179,18 +183,30 @@ func (i *httpImage) Import(copier func(io.Reader) error, vol libvirtxml.StorageV
 	if vol.Target.Timestamps != nil && vol.Target.Timestamps.Mtime != "" {
 		req.Header.Set("If-Modified-Since", timeFromEpoch(vol.Target.Timestamps.Mtime).UTC().Format(http.TimeFormat))
 	}
-	response, err := client.Do(req)
 
-	if err != nil {
-		return fmt.Errorf("Error while downloading %s: %s", i.url.String(), err)
-	}
+	var response *http.Response
+	for retryCount := 0; retryCount < maxHTTPRetries; retryCount++ {
+		response, err = client.Do(req)
+		if err != nil {
+			return fmt.Errorf("Error while downloading %s: %v", i.url.String(), err)
+		}
+		defer response.Body.Close()
 
-	defer response.Body.Close()
-	if response.StatusCode == http.StatusNotModified {
-		return nil
+		log.Printf("[DEBUG]: url resp status code %s (retry #%d)\n", response.Status, retryCount)
+		if response.StatusCode == http.StatusNotModified {
+			return nil
+		} else if response.StatusCode == http.StatusOK {
+			return copier(response.Body)
+		} else if response.StatusCode < 500 {
+			break
+		} else {
+			// retry
```

Comment:
for this comment here i would expect that either we add more value to it, saying dunno why we are retrying and which behaviour this can happen like `Retry HTTP downloads for 5xx errors` or we can just omit ( because a programmer could read the `retrycount` var) and directly thinking there is a retry